### PR TITLE
Add create and update support for configmap resource

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#4845

### CODE REVIEW

File: service/controller/app/v1/resource/configmap/create.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		_, err = ctlCtx.K8sClient.CoreV1().ConfigMaps(configMap.Namespace).Create(configMap)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensured creation of configmap %#q", configMap.Name))
+	} else {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("no need to create configmap"))
+	}
+	return nil
```
Comment (xh3b4sd) [inline service/controller/app/v1/resource/configmap/create.go]:
```suggestion

	return nil
```

File: service/controller/app/v1/resource/configmap/create_test.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+					Namespace: "default",
+				},
+			},
+			expectedConfigMap: &corev1.ConfigMap{},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			c := Config{
```
Comment (tomahawk28) [inline service/controller/app/v1/resource/configmap/create_test.go]: I think we can move `c` out of this loop, right?

**Comment by rossf7** [comment]
> We could maybe come up with a general one.

I like the idea but there is some app-operator specific stuff. We need to use k8s clients from the context because the operator needs to connect to control plane and tenant clusters.

My concern is more with the desired state. We need merge logic that is quite complex and basically the same for configmaps and secrets. We also need to merge the configmap and secret values in chart-operator. 

I'm going to add a `values` package in app-operator but that could become a library.

File: service/controller/app/v1/resource/configmap/create_test.go (hunk 2 above)
Comment (rossf7) [reply service/controller/app/v1/resource/configmap/create_test.go]: Thanks good point. Done.

**Comment by rossf7** [comment]
Going with this. I'll keep in mind if we can do something with more generic k8s resources.

**Pull request merged**